torch_xla/csrc/tensor_util.cpp

```diff
@@ -651,9 +651,11 @@ void PopulateTensorBuffer(const at::Tensor& tensor,
 
 torch::lazy::BackendDataPtr TensorToXlaData(
     const at::Tensor& tensor, const xla::Shape& shape,
-    const torch::lazy::BackendDevice& device, bool transfer_async) {
+    const torch::lazy::BackendDevice& device) {
   XLA_TIMED("TensorToData");
-  if (transfer_async) {
+  static const bool transfer_async =
+      xla::sys_util::GetEnvBool("XLA_TRANSFER_SCALAR_ASYNC", false);
+  if (tensor.dim() == 0 && tensor.numel() == 1 && transfer_async) {
```

Comment:
nit, can we move `transfer_async` check to first since we expect this to be false in may cases.

Reply:
Fixed!